crates/transaction-pool/src/blobstore/disk.rs

```diff
@@ -52,22 +53,33 @@ impl BlobStore for DiskFileBlobStore {
 
     fn insert_all(&self, txs: Vec<(B256, BlobTransactionSidecar)>) -> Result<(), BlobStoreError> {
         if txs.is_empty() {
-            return Ok(())
+            return Ok(());
         }
         self.inner.insert_many(txs)
     }
 
     fn delete(&self, tx: B256) -> Result<(), BlobStoreError> {
-        self.inner.delete_one(tx)?;
+        let path = self.inner.blob_disk_file(tx);
+        self.files_to_delete.write().insert(path);
         Ok(())
     }
 
     fn delete_all(&self, txs: Vec<B256>) -> Result<(), BlobStoreError> {
-        if txs.is_empty() {
-            return Ok(())
+        let paths = txs.into_iter().map(|tx| self.inner.blob_disk_file(tx)).collect::<HashSet<_>>();
+        let mut files_to_delete = self.files_to_delete.write();
+        for path in paths {
+            files_to_delete.insert(path);
         }
+        Ok(())
+    }
 
-        self.inner.delete_many(txs)?;
+    fn cleanup(&self) -> Result<(), BlobStoreError> {
+        let files_to_delete = self.files_to_delete.write().drain().collect::<Vec<_>>();
+        for file_path in files_to_delete {
+            if let Err(e) = std::fs::remove_file(&file_path) {
+                return Err(BlobStoreError::Other(Box::new(e)));
+            }
```

Comment:
here we don't want to return early because we need to clean up all files, instead we should just log the error on debug!

Reply:
I changed this to print a debug and remove files in the said path
```
for file_path in files_to_delete {
            if let Err(e) = std::fs::remove_file(&file_path) {
                debug!("Failed to delete file {}: {:?}", file_path.display(), e);
            }
        }
```